### PR TITLE
Removed the redundant data 

### PR DESCRIPTION
**Pull request by rsarwad**
Removed the redundant data from structures, ue_mm_context_t and emm_context_t

(cherry picked from commit bcdc5117f2bc3d1bfab6f6748417310b303c3a2d)

### CODE REVIEW

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 1]
```diff
@@ -341,11 +341,6 @@ typedef struct ue_mm_context_s {
   /* Basic identifier for ue. IMSI is encoded on maximum of 15 digits of 4 bits,
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h]: Remove this comment as well

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 2]
```diff
@@ -341,11 +341,6 @@ typedef struct ue_mm_context_s {
...
    * so usage of an unsigned integer on 64 bits is necessary.
    */
-  imsi64_t imsi; // set by nas_auth_param_req_t
-  uint8_t imsi_len;
-  uint8_t member_present_mask; // bitmask, see significance of bits below
-#define UE_CTXT_MEMBER_IMEI_SV (1 << 0)
-  imeisv_t imeisv; // IMEISV received in Security mode complete
   // int                    len;
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h]: Remove this commented line

File: lte/gateway/c/oai/tasks/nas/emm/NasTransportHdl.c [hunk 3]
```diff
@@ -116,6 +116,8 @@ int emm_proc_uplink_nas_transport(mme_ue_s1ap_id_t ue_id, bstring nas_msg_pP)
   ue_ctx_p = emm_context_get(&_emm_data, ue_id);
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/nas/emm/NasTransportHdl.c]: Can we rename this to emm_ctxt_p as it is a pointer to the emm_context?

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 4]
```diff
@@ -369,8 +360,6 @@ typedef struct ue_mm_context_s {
 
   /* TODO: Add TAI list */
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h]: This TODO is not needed any more.

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 5]
```diff
@@ -369,8 +360,6 @@ typedef struct ue_mm_context_s {
...
   tai_t serving_cell_tai;
-  tai_list_t tail_list; // Current Tracking area list
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h:372]: Let's keep all the TAI related context together, either all in ue_mm_context_t or all in emm_context_t.

File: lte/gateway/c/oai/include/mme_app_ue_context.h (hunk 4 above)
Comment (ssanadhya) [reply lte/gateway/c/oai/include/mme_app_ue_context.h]: The commented code above can also be removed as part of this change.

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 6]
```diff
@@ -385,68 +363,22 @@ typedef struct ue_mm_context_s {
...
-  // Selected CN operator id      // Selected core network operator identity (to support network sharing as defined in TS 23.251 [24]).
-
-  // Recovery                     // Indicates if the HSS is performing database recovery.
-
   ard_t
     access_restriction_data; // The access restriction subscription information. set by S6A UPDATE LOCATION ANSWER
 
-  // ODB for PS parameters        // Indicates that the status of the operator determined barring for packet oriented services.
-
   // APN-OI Replacement           // Indicates the domain name to replace the APN-OI when constructing the PDN GW
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h:372]: Convert this and next few lines into one multi-line comment describing apn_oi_replacement

File: lte/gateway/c/oai/include/mme_app_ue_context.h [hunk 7]
```diff
@@ -385,68 +363,22 @@ typedef struct ue_mm_context_s {
...
   // FQDN upon which to perform a DNS resolution. This replacement applies for all
   // the APNs in the subscriber's profile. See TS 23.003 [9] clause 9.1.2 for more
   // information on the format of domain names that are allowed in this field.
   bstring apn_oi_replacement; // example: "province1.mnc012.mcc345.gprs"
 
-  // MME IP address for S11       // MME IP address for the S11 interface (used by S-GW)
-  // LOCATED IN mme_config_t.ipv4.s11
-
   // MME TEID for S11             // MME Tunnel Endpoint Identifier for S11 interface.
```
Comment (ssanadhya) [inline lte/gateway/c/oai/include/mme_app_ue_context.h:378]: Combine this and next line into one multi-line comment describing mme_teid_s11

File: lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp [hunk 8]
```diff
@@ -236,14 +236,8 @@ void MmeNasStateConverter::ue_context_to_proto(
...
 
-  ue_ctxt_proto->set_imsi(ue_ctxt->imsi);
-  ue_ctxt_proto->set_imsi_len(ue_ctxt->imsi_len);
-  ue_ctxt_proto->set_member_present_mask(ue_ctxt->member_present_mask);
-
-  memcpy(
-    ue_ctxt_proto->mutable_imeisv(),
-    &ue_ctxt->imeisv.u.value,
-    IMEISV_BCD8_SIZE);
+  ue_ctxt_proto->set_imsi(ue_ctxt->emm_context._imsi64);
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:239]: The UE context proto message does not have any member field imsi or imsi_len any more. So lines 239 and 240 are not needed.

File: lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp [hunk 9]
```diff
@@ -284,13 +267,8 @@ void MmeNasStateConverter::proto_to_ue_mm_context(
...
   ue_mm_context_t* state_ue_mm_context)
 {
-  state_ue_mm_context->imsi = ue_context_proto->imsi();
-  state_ue_mm_context->imsi_len = ue_context_proto->imsi_len();
-  state_ue_mm_context->member_present_mask =
-    ue_context_proto->member_present_mask();
-  strcpy(
-    (char*) state_ue_mm_context->imeisv.u.value,
-    ue_context_proto->imeisv().c_str());
+  state_ue_mm_context->emm_context._imsi64 = ue_context_proto->imsi();
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:270]: The state_ue_mm_context->emm_context struct is updated via the emm_context_to_proto function call. So lines 270 and 271 are not needed.

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp (hunk 8 above)
Comment (rsarwad) [reply lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:239]: Addressed in PR 766

File: lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp (hunk 9 above)
Comment (rsarwad) [reply lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:270]: Updated in PR 766

File: lte/gateway/c/oai/include/mme_app_ue_context.h (hunk 4 above)
Comment (rsarwad) [reply lte/gateway/c/oai/include/mme_app_ue_context.h]: Removed in PR, 766

File: lte/gateway/c/oai/include/mme_app_ue_context.h (hunk 6 above)
Comment (rsarwad) [reply lte/gateway/c/oai/include/mme_app_ue_context.h:372]: Done in PR 766

File: lte/gateway/c/oai/include/mme_app_ue_context.h (hunk 7 above)
Comment (rsarwad) [reply lte/gateway/c/oai/include/mme_app_ue_context.h:378]: Done in PR, 766

**Comment by rsarwad** [comment]
The changes are available as part of PR, 766

**Pull request closed without merging**